site/src/components/Filter/SelectFilter.tsx

```diff
@@ -0,0 +1,117 @@
+import visuallyHidden from "@mui/utils/visuallyHidden";
+import { useState, type FC, type ReactNode } from "react";
+import { Loader } from "components/Loader/Loader";
+import {
+  SelectMenu,
+  SelectMenuTrigger,
+  SelectMenuButton,
+  SelectMenuContent,
+  SelectMenuSearch,
+  SelectMenuList,
+  SelectMenuItem,
+  SelectMenuIcon,
+} from "components/SelectMenu/SelectMenu";
+
+const BASE_WIDTH = 200;
+const POPOVER_WIDTH = 320;
+
+export type SelectFilterOption = {
+  startIcon?: ReactNode;
+  label: string;
+  value: string;
+};
+
+export type SelectFilterProps = {
+  options: SelectFilterOption[] | undefined;
+  selectedOption?: SelectFilterOption;
+  // Used to add a accessibility label to the select
+  label: string;
+  // Used when there is no option selected
+  placeholder: string;
+  // Used to customize the empty state message
+  emptyText?: string;
+  onSelect: (option: SelectFilterOption | undefined) => void;
+  // SelectFilterSearch element
+  search?: ReactNode;
+};
+
+export const SelectFilter: FC<SelectFilterProps> = ({
+  label,
+  options,
+  selectedOption,
+  onSelect,
+  placeholder,
+  emptyText,
+  search,
+}) => {
+  const [open, setOpen] = useState(false);
+
+  return (
+    <SelectMenu open={open} onOpenChange={setOpen}>
+      <SelectMenuTrigger>
+        <SelectMenuButton
+          startIcon={selectedOption?.startIcon}
+          css={{ width: BASE_WIDTH }}
+        >
+          {selectedOption?.label ?? placeholder}
+          <span css={{ ...visuallyHidden }}>{label}</span>
```

Comment:
I didn't notice it 🤦 My idea is, since this is a select component, I think having a label like "Select a user" is better than having the selected value as the label "User 2" for example.

Reply:
I used `aria-label` in the button directly.